### ISSUE TITLE
upgrade dependency proc-macro2

### ISSUE THREAD

**Issue by jtmoon79** [frondeus/test-case#129]
tl;dr upgrading `proc-macro2` version would avoid a known `proc-macro2` bug `error[E0635]: unknown feature proc_macro_span_shrink`

<br/>

[My project](https://github.com/jtmoon79/super-speedy-syslog-searcher/blob/0.6.68/Cargo.toml#L107) is using `test_case` v3.1.0 which depends on `proc-macro2` v1.0.52.

```text
$ cargo tree
...
└── test-case v3.1.0
    └── test-case-macros v3.1.0 (proc-macro)
        ├── proc-macro-error v1.0.4
        │   ├── proc-macro-error-attr v1.0.4 (proc-macro)
        │   │   ├── proc-macro2 v1.0.52 (*)
        │   │   └── quote v1.0.26 (*)
        │   │   [build-dependencies]
        │   │   └── version_check v0.9.4
        │   ├── proc-macro2 v1.0.52 (*)
        │   ├── quote v1.0.26 (*)
        │   └── syn v1.0.109 (*)
        │   [build-dependencies]
        │   └── version_check v0.9.4
        ├── proc-macro2 v1.0.52 (*)
        ├── quote v1.0.26 (*)
        ├── syn v1.0.109 (*)
        └── test-case-core v3.1.0
            ├── cfg-if v1.0.0
            ├── proc-macro-error v1.0.4 (*)
            ├── proc-macro2 v1.0.52 (*)
            ├── quote v1.0.26 (*)
            └── syn v1.0.109 (*)
```

`proc-macro2` Issue https://github.com/dtolnay/proc-macro2/issues/356 causes failures of some tools like [`udeps`](https://github.com/est31/cargo-udeps) with error
```
$ cargo +nightly udeps
error[E0635]: unknown feature `proc_macro_span_shrink`
  --> /home/ulug/.cargo/registry/src/index.crates.io-6f17d22bba15001f/proc-macro2-1.0.52/src/lib.rs:92:30
   |
92 |     feature(proc_macro_span, proc_macro_span_shrink)
   |                              ^^^^^^^^^^^^^^^^^^^^^^
```

According to the linked `proc-macro2` Issue

> you need to upgrade proc-macro2 to 1.0.60+, downgrade proc-macro2 to 1.0.43 or before, downgrade the nightly toolchain to nightly-2023-06-28 or before, or use a non-nightly toolchain.

This error occurred on rust 1.70.0 and 1.72.0.

Can `test-case` upgrade it's `proc-macro2` dependency version?

**Comment by luke-biel** [comment]
Sorry for getting back to you so late. This is not a test-case issue, as we don't provide `Cargo.lock` with our crate (and I think these are not used when compiling dependency libraries anyway). You should update your project, our lock version `1.0` is compatible with everything that dtolnay released so far. Try

`cargo update -p proc-macro2`
